### PR TITLE
feat: add nodemon, eslint, gitignore, jest, ts, package.json, CHANGELOG and test CI with helloWorld function

### PR DESCRIPTION
**Pull request by vitordullens**
#### What is the purpose of this pull request?

Create all the config files necessary to start a TypeScript Project and test the CI configuration by adding a helloWorld function.

#### What problem is this solving?

Before it there were no config files, so it wasn't possible to start a project and it was impossible to know if the CI was working properly.

#### Types of changes
- [ ] Refactor (non-breaking change that only makes the code better)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const helloWorld = (msg : string = 'Hello World!') => {
```
Comment (VerasThiago) [inline src/index.ts:1]: The ideia in the `index` is to import the `helloWorld` function from `modules` folder so then you can call `helloWorld()`

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  process.stdout.write(msg);
+  return msg;
+};
+
+helloWorld();
+
+export default helloWorld;
```
Comment (VerasThiago) [inline src/index.ts:8]: You export this function to where ? No one is using it.

Comment (VerasThiago) [inline src/index.ts:8]: Also take a look of the difference between `export default` and `export` and explain here, please.

**Comment by VerasThiago** [comment]
You need to set this repo to require at lease 1 review.

**Comment by VerasThiago** [comment]
Where is `CHANGELOG.md` file ?

File: src/index.ts (hunk 2 above)
Comment (vitordullens) [reply src/index.ts:8]: I needed to export this function so that it could be used in the test file.

File: src/index.ts (hunk 1 above)
Comment (vitordullens) [reply src/index.ts:1]: okay 👍

**Comment by vitordullens** [comment]
> You need to set this repo to require at lease 1 review.
> 
>

File: src/index.ts (hunk 2 above)
Comment (vitordullens) [reply src/index.ts]: With `export default` the naming of the import is independent, so we can use any name we like, it's the default "thing" imported, so in this case, we could import helloWorld using `import hello from '../modules/helloWorld'`. A single file can only have one default exporting. However with only `export` we can export more than one function/class/const(...) but the name will be the same as the name in the respective file, i.e: `import { helloWorld } from '../modules/helloWolrd'`.

So basically `export default` is useful when exporting only one object and `export` is useful when you need to export several values.

Comment (VerasThiago) [reply src/index.ts]: Awesome, so in this case, we want to `export default` or `export` ? (tell me why)

Comment (vitordullens) [reply src/index.ts]: In this case the ideal would be to use `export default`, because we only want to export **one** function, that is the core function of the file. 😀
